frontend/src/EventDetails.jsx

```diff
@@ -1,67 +1,87 @@
-import React from 'react';
+import React, { useState, useEffect } from "react";
 import './EventDetails.css';
 import backButton from './return-button.png';
+import { Link, useParams } from 'react-router-dom';
 import dummyPoster from './dummy-poster.png';
 import NavigationBar from './NavigationBar';
 
 const EventDetails = () => {
-    const dummyDescription = "Join us for a night of fun and games! Variety of card and board games to play! Bring your friends and/or have the opportunity to meet some new people!";
 
-    
+    const [event, setEvent] = useState([]);
+
+    let {id} = useParams();
+    console.log(id);
+
+    useEffect(() => {
+        // Function to fetch events from the API
+        const fetchEventByID = async () => {
+        try {
+            const response = await fetch(`http://localhost:8080/events/${id}`);
```

Comment:
Let's get the backend server URL from an environment file to make this code more generalisable.